avaje-config/src/main/java/io/avaje/config/InitialLoader.java

```diff
@@ -204,6 +205,19 @@ private void loadViaIndirection() {
     }
   }
 
+  /** Load configuration defined by a <em>config.profiles</em> property. */
+  private void loadViaProfiles() {
+    final String paths = loadContext.profiles();
+    if (paths != null) {
+      for (final String path : splitPaths(paths)) {
+        final var profile = loadContext.eval(path);
+        loadWithExtensionCheck("application-" + profile + ".properties");
+        loadWithExtensionCheck("application-" + profile + ".yaml");
+        loadWithExtensionCheck("application-" + profile + ".yml");
```

Comment:
I'll merge this in but I think I'm noticing something subtle about loadWithExtensionCheck() ... which is that it will not load the FILE if the RESOURCE is found.

So I'm pondering:
- For this case we might want to try and load RESOURCE **_and_** FILE (regardless if the RESOURCE is found)
- Ordering wise we might want to load all the RESOURCE for all the profiles first, and then load all the FILE for all the profiles after (ordering here in terms of file properties overriding resource properties).

So I'm pondering those 2 subtle things.  I'll merge this into master now though, get a new coffee and ponder those 2 things ...

Reply:
typically, my property files are only in `src/main/resources` so I'm not bothered

Reply:
I thought that was a feature tbh

Reply:
Yeah, I get that. 99% of the time there will not be FILE & RESOURCE defined (but for that 1% case I think we need to be consistent. Config generally loads all the FILEs afterwards as those are the _"overrides"_ to the RESOURCE _"defaults"._

Reply:
> I thought that was a feature tbh

Yeah, might be a bug in my eyes :)

That is, when we have a properties/yaml as FILE (so externally supplied) in theory it's put there to override RESOURCE (application supplied defaults) properties.

So Config as I see it really should be loading any FILE, and loading it after RESOURCE properties have been loaded. So it might be a feature but right now I feel it might be more a subtle "historic bug". It might be ok because the use cases typically do not have **_both_** RESOURCE and FILE.  

It is tempting to change that to try to load FILE regardless if there was a RESOURCE found and loaded.  Pondering, Hmm ...